torch/utils/_python_dispatch.py

```diff
@@ -3,11 +3,13 @@
 import functools
 
 import warnings
-from torch.utils._mode_utils import _enable_mode, _ModeInfo, _restore_mode
+from torch.utils._mode_utils import _enable_mode_checks, _ModeInfo
 from torch._C import _get_torch_dispatch_mode, _set_torch_dispatch_mode
 from dataclasses import dataclass
 
 
+_cur_torch_dispatch_mode = []
```

Comment:
Might be worth thinking about if this should be global or thread-local. FWIW functorch's stack is thread-local, because autograd has multiple threads of execution in the backward pass